presenters/powder_projection_presenter.py

```diff
@@ -1,25 +1,28 @@
-from mainview import MainView
 from models.projection.powder.projection_calculator import ProjectionCalculator
 from views.powder_projection_view import PowderView
 from widgets.projection.powder.command import Command
+from validation_decorators import require_main_presenter
+from mainview import MainView
 
 
 class PowderProjectionPresenter(object):
 
-    def __init__(self,powder_view,main_view,projection_calculator):
+    def __init__(self, powder_view, projection_calculator):
         self._powder_view = powder_view
         self._projection_calculator = projection_calculator
         if not isinstance(self._powder_view,PowderView):
             raise TypeError("powder_view is not of type PowderView")
         if not isinstance(self._projection_calculator,ProjectionCalculator):
             raise TypeError("projection_calculator is not of type ProjectionCalculator")
-        if not isinstance(main_view,MainView):
-            raise TypeError("main_view is not of type MainView")
 
-        self._main_view = main_view
         #Add rest of options
         self._powder_view.populate_powder_u1(['|Q|'])
         self._powder_view.populate_powder_u2(['Energy'])
+        self._main_presenter = None
+
+    def register_master(self, main_view):
```

Comment:
I think this argument should be the `presenter` not the `view`. @OwenArnold, do you agree?